src/test/java/com/sigma/cinematicketpro/TestUtils.java

```diff
@@ -0,0 +1,72 @@
+package com.sigma.cinematicketpro;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.sigma.cinematicketpro.dto.MovieDto;
+import com.sigma.cinematicketpro.entity.Movie;
+import com.sigma.cinematicketpro.mapper.MovieMapper;
+import com.sigma.cinematicketpro.mapper.impl.MovieMapperImpl;
+import org.modelmapper.ModelMapper;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.io.UnsupportedEncodingException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+
+public final class TestUtils {
+    private final static ObjectMapper objectMapper = getObjectMapper();
+
+
+    private final static MovieMapper movieMapper = new MovieMapperImpl(new ModelMapper());
+
+    private TestUtils() {
+        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
+    }
+
+    public static ObjectMapper getObjectMapper() {
+        return Optional.ofNullable(objectMapper)
+                .orElseGet(() -> {
+                    ObjectMapper mapper = new ObjectMapper();
+                    mapper.registerModule(new JavaTimeModule());
+                    return mapper;
+                });
+    }
+
+    public static List<Movie> getValidMovies() {
+        return List.of(
+                Movie.builder()
+                        .id("10020")
+                        .tmdbId(238L)
+                        .title("The Godfather")
+                        .voteAverage(8.7)
+                        .voteCount(18200L)
+                        .overview("Spanning the years 1945 to 1955...")
+                        .posterPath("/3bhkrj58Vtu7enYsRolD1fZdja1.jpg")
+                        .releaseDate(LocalDate.parse("1972-03-14", DateTimeFormatter.ofPattern("yyyy-MM-dd")))
+                        .build(),
+                Movie.builder()
+                        .id("someId")
+                        .tmdbId(238123L)
+                        .title("Fast X")
+                        .voteAverage(7.3)
+                        .voteCount(1848L)
+                        .overview("Over many missions and against...")
+                        .posterPath("/fiVW06jE7z9YnO4trhaMEdclSiC.jpg")
+                        .releaseDate(LocalDate.parse("2023-05-17", DateTimeFormatter.ofPattern("yyyy-MM-dd")))
+                        .build()
+        );
+    }
+
+    public static <T> T mapMvcResultIntoObject(MvcResult mvcResult, TypeReference<T> typeReference) throws JsonProcessingException, UnsupportedEncodingException {
```

Comment:
why do you need generics here? generics is Object class technicly
just map MvcResult into JSONObject and assertJSON, or just compare strings - actual outcome vs expected

Reply:
Generics are used to prevent explicit type conversion. 
As I see it, there are two main ways to solve this problem. The first is to map MvcResult to JSONObject, map the entity to JSONObject and compare them. The second is to map MvcResult to request result. So I chose the second option because it looks more convenient to map MvcResult to an object instead of mapping both the entity and MvcResult to a JSONObject.